libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -160,14 +161,14 @@ class FinalChainImpl final : public FinalChain {
       precommit_ext(*result, batch);
     }
     db_->commitWriteBatch(batch, db_opts_w_);
+    block_finalized_emitter_.emit(result);
     state_api_.transition_state_commit();
     {
       unique_lock l(last_block_mu_);
       last_block_ = blk_header;
     }
     num_executed_dag_blk_ = num_executed_dag_blk;
     num_executed_trx_ = num_executed_trx;
-    block_finalized_emitter_.emit(result);
```

Comment:
you sure about this change? shouldn't we wait for `transition_state_commit`

Reply:
It works fine, but I just forgot to move it back after tries to apply hardfork also on that event(to not create new one). Will move it back